winter/http/media_type.py

```diff
@@ -0,0 +1,125 @@
+from typing import Dict
+
+from typing import Tuple
+
+
+class InvalidMediaTypeException(Exception):
+    def __init__(self, media_type: str, message: str):
+        super().__init__(f'{message}: {media_type}')
+        self.media_type = media_type
+        self.message = message
+
+
+class MediaType:
+    ALL: 'MediaType' = None
+    APPLICATION_ATOM_XML: 'MediaType' = None
+    APPLICATION_FORM_URLENCODED: 'MediaType' = None
+    APPLICATION_JSON: 'MediaType' = None
+    APPLICATION_JSON_UTF8: 'MediaType' = None
+    APPLICATION_OCTET_STREAM: 'MediaType' = None
+    APPLICATION_PDF: 'MediaType' = None
+    APPLICATION_PROBLEM_JSON: 'MediaType' = None
+    APPLICATION_PROBLEM_JSON_UTF8: 'MediaType' = None
+    APPLICATION_PROBLEM_XML: 'MediaType' = None
+    APPLICATION_RSS_XML: 'MediaType' = None
+    APPLICATION_STREAM_JSON: 'MediaType' = None
+    APPLICATION_XHTML_XML: 'MediaType' = None
+    APPLICATION_XML: 'MediaType' = None
+    IMAGE_GIF: 'MediaType' = None
+    IMAGE_JPEG: 'MediaType' = None
+    IMAGE_PNG: 'MediaType' = None
+    MULTIPART_FORM_DATA: 'MediaType' = None
+    TEXT_EVENT_STREAM: 'MediaType' = None
+    TEXT_HTML: 'MediaType' = None
+    TEXT_MARKDOWN: 'MediaType' = None
+    TEXT_PLAIN: 'MediaType' = None
+    TEXT_XML: 'MediaType' = None
+
+    def __init__(self, media_type_str: str):
+        self._type_ = '*'
+        self._subtype = '*'
+        self._parameters: Dict[str, str] = {}
```

Comment:
I think it is better to use tuple of MediaTypeParameter like so
```

@dataclass(frozen=True)
class MediaTypeParameter:
    name: str
    value: str

def __str__(self):
    return f'{self.name}={self.value}'


```

Reply:
@berenda-andrey MediaType has some pre-defined set of possible parameters, like charset. dict will help to lookup for these fields: `MediaType('test/plain; charset=utf-8').parameters.get('charset')`. Could you provide somw benefits of using `List[MediaTypeParameter]`?

Reply:
It was immutable and hashable by default
I think one media type have not many parameters and searching in typle is also fast

Reply:
not list, tuple